src/main/java/seedu/recipe/model/recipe/ingredient/IngredientBuilder.java

```diff
@@ -0,0 +1,128 @@
+package seedu.recipe.model.recipe.ingredient;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.recipe.commons.util.AppUtil.checkArgument;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.AMOUNT_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.COMMON_NAME_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.ESTIMATE_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.NAME_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.REMARK_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.SUBSTITUTION_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.parse;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import seedu.recipe.logic.parser.Prefix;
+
+/**
+ * Represents a recipe's ingredient in the recipe book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidIngredient(String)}
+ */
+public class IngredientBuilder {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ingredients should follow this format: \n"
+            + "`[-a AMOUNT] [-e ESTIMATED AMOUNT] -n NAME `"
```

Comment:
Amount and estimated amount feel somewhat similar but it should be fine to keep.